include/fiction/networks/virtual_pi_network.hpp

```diff
@@ -0,0 +1,505 @@
+
+            return children;
+        };
+
+#if (PROGRESS_BARS)
+        // initialize a progress bar
+        mockturtle::progress_bar bar{static_cast<uint32_t>(ntk.num_gates()), "[i] network conversion: |{0}|"};
+#endif
+
+        ntk_topo.foreach_gate(
+            [&, this](const auto& g, [[maybe_unused]] auto i)
+            {
+                auto children = gather_fanin_signals(g);
+
+#if (PROGRESS_BARS)
+                // update progress
+                bar(i);
+#endif
+
+                if constexpr (mockturtle::has_is_and_v<Ntk> && mockturtle::has_create_and_v<Ntk>)
+                {
+                    if (ntk.is_and(g))
+                    {
+                        old2new[g] = ntk_dest.create_and(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_or_v<Ntk> && mockturtle::has_create_or_v<Ntk>)
+                {
+                    if (ntk.is_or(g))
+                    {
+                        old2new[g] = ntk_dest.create_or(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_xor_v<Ntk> && mockturtle::has_create_xor_v<Ntk>)
+                {
+                    if (ntk.is_xor(g))
+                    {
+                        old2new[g] = ntk_dest.create_xor(children[0], children[1]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_is_maj_v<Ntk> && mockturtle::has_create_maj_v<Ntk>)
+                {
+                    if (ntk.is_maj(g))
+                    {
+                        old2new[g] = ntk_dest.create_maj(children[0], children[1], children[2]);
+                        return true;
+                    }
+                }
+                if constexpr (fiction::has_is_buf_v<Ntk> && mockturtle::has_create_buf_v<Ntk>)
+                {
+                    if (ntk.is_buf(g))
+                    {
+                        old2new[g] = ntk_dest.create_buf(children[0]);
+                        return true;
+                    }
+                }
+                if constexpr (mockturtle::has_node_function_v<Ntk> && mockturtle::has_create_node_v<Ntk>)
+                {
+                    old2new[g] = ntk_dest.create_node(children, ntk.node_function(g));
+                    return true;
+                }
+
+                return true;
+            });
+
+        ntk.foreach_po(
+            [this, &ntk_dest, &old2new](const auto& po)
+            {
+                const auto tgt_signal = old2new[ntk.get_node(po)];
+                const auto tgt_po     = ntk.is_complemented(po) ? ntk_dest.create_not(tgt_signal) : tgt_signal;
+
+                ntk_dest.create_po(tgt_po);
+            });
+
+        // restore signal names if applicable
+        fiction::restore_names(ntk, ntk_dest, old2new);
```

Comment:
```suggestion
        restore_names(ntk, ntk_dest, old2new);
```
should work or did you encounter issues on Windows?

Reply:
Should be fine like this :)